src/runtime_src/core/tools/xbutil2/SubCmdValidate.cpp

```diff
@@ -424,22 +424,35 @@ free_unmap_bo(xclDeviceHandle handle, xclBufferHandle boh, void * boptr, size_t
  * helper function for P2P test
  */
 static bool
-p2ptest_set_or_cmp(char *boptr, size_t size, char pattern, bool set)
+p2ptest_set_or_cmp(char *boptr, size_t size, char *valid_data, size_t valid_data_size, bool set)
```

Comment:
The code below is using a vector of chars.  Instead of breaking in to a buffer and size, having an argument (e.g., `std::vector<char> & validData`), would be a better and more scalable solution. 

```suggestion
  p2ptest_set_or_cmp(char *boptr, size_t size, const std::vector<char> & validData, bool set)
```